lib/Driver/Driver.cpp

```diff
@@ -2786,6 +2789,41 @@ void Driver::chooseSwiftModuleOutputPath(Compilation &C,
   }
 }
 
+void Driver::choosePrivateOutputFilePath(Compilation &C,
+                                         const TypeToPathMap *OutputMap,
+                                         StringRef workingDirectory,
+                                         CommandOutput *Output,
+                                         file_types::ID fileID) const {
+  if (hasExistingAdditionalOutput(*Output, fileID))
+    return;
+
+  StringRef OFMOutputPath;
+  if (OutputMap) {
+    auto iter = OutputMap->find(fileID);
+    if (iter != OutputMap->end())
+      OFMOutputPath = iter->second;
+  }
+  if (!OFMOutputPath.empty()) {
+    // Prefer a path from the OutputMap.
+    Output->setAdditionalOutputForType(fileID, OFMOutputPath);
+  } else if (Output->getPrimaryOutputType() != file_types::TY_Nothing) {
+    auto ModulePath = Output->getAnyOutputForType(file_types::TY_SwiftModuleFile);
+    bool isTempFile = C.isTemporaryFile(ModulePath);
+    auto ModuleName = llvm::sys::path::filename(ModulePath);
+    llvm::SmallString<128> Path(llvm::sys::path::parent_path(ModulePath));
+    llvm::sys::path::append(Path, "Private");
+    // If the build system has created a Private dir for us to include the file, use it.
+    if (!llvm::sys::fs::exists(Path)) {
+      llvm::sys::path::remove_filename(Path);
```

Comment:
Ah, clever. And I guess we can look in both places?

Reply:
yeah, the source info loader should look into both places and prefer the `Private` one.